### DISCUSSION TITLE
Add color for bed track

### DISCUSSION THREAD

**Discussion by lalalagartija** [GMOD/jbrowse-components#2772] [Q&A] [answered]
Hi, 
I am trying to add colors to a bigbed track without sucess. 
I tried  
`jbrowse add-track file.bigbed -a genome --load copy --out /var/www/html/jbrowse2/ --config '{"displays": [{"displayId": "basicdisplay","type":"LinearBasicDisplay", "renderer": {"color1": "green", "type": "SvgFeatureRenderer"}}]}' `
and 
`jbrowse add-track file.bigbed -a genome --load copy --out /var/www/html/jbrowse2/ --config '{"displays": [{"renderer": {"color1": "green"}}]}'`

none of them worked
Thank you,

**Comment by garrettjstevens** [comment]
Hi @lalalagartija,

I tried using the same `--config` as your first example, and it worked for me, so I'm not sure what's going on. Could you provide some more information about what you see? For example, a screenshot of your track and/or a copy of the track config that gets added to your "config.json".

For reference, this is what I see when running that command (modified to use my own bigbed file):
![image](https://user-images.githubusercontent.com/25592344/155553745-ba9c93ab-81cc-432c-8dd2-70b6e6274be6.png)

**Comment by lalalagartija** [comment]
Hi, 
Thank you for your answer. 
Actually I deleted the old files because I want to get different colors for each gene. I tried a new method that is still unsucessfull..
./bedToBigBed -as=blast_clusters.as2 -type=bed9+2 -extraIndex=nothing,Cluster blast_clusters.bed9 chrom.sizes  myBigBed2.bb
browse add-track myBigBed.bb -a Virus1 --load copy --out /var/www/html/viruses/ --force

The resulting track on Jbrowse2 is empty. I can't see any of the bed features.
I used a .as file that looks like this : 
"
table name
"UCSC Genes for Pithovirus_massiliensis with color plus"
(
string  chrom;		"Reference sequence chromosome or scaffold"
uint    chromStart;	"Start position of feature on chromosome"
uint    chromEnd;	"End position of feature on chromosome"
string  name;		"Name of gene"
uint    score;		"Score"
char[1] strand;		"+ or - for strand"
uint    thickStart;	"Coding region start"
uint    thickEnd;	"Coding region end"
uint  	reserved;	"Green on + strand, Red on - strand"
string  nothing;	"Gene Symbol"
string  Cluster;		"SWISS-PROT protein Accession number"
)
"
bed file : 
"
Virus1	22888	23064	name1	0	+	22888	23064	60,18,208	.	Cluster1
Virus1	121877	122058	name2	0	+	121877	122058	60,18,208	.	Cluster1
Virus1	181522	181702	name3	0	+	181522	181702	60,18,208	.	Cluster1
Virus1	199225	199406	name4	0	+	199225	199406	60,18,208	.	Cluster1
Virus1	199575	199755	name5	0	+	199575	199755	60,18,208	.	Cluster1
Virus1	199856	200035	name6	0	-	199856	200035	60,18,208	.	Cluster1
Virus1	238392	238573	name7	0	+	238392	238573	60,18,208	.	Cluster1
Virus1	286233	286409	name8	0	-	286233	286409	60,18,208	.	Cluster1
Virus1	545032	545210	name9	0	-	545032	545210	60,18,208	.	Cluster1
Virus1	620602	620784	name10	0	+	620602	620784	60,18,208	.	Cluster1

"
And the config.json for this track is : 
"
 {
      "type": "FeatureTrack",
      "trackId": "myBigBed",
      "name": "myBigBed",
      "adapter": {
        "type": "BigBedAdapter",
        "bigBedLocation": {
          "uri": "myBigBed.bb",
          "locationType": "UriLocation"
        }
      },
      "assemblyNames": [
        "Virus1"
      ]
    }
"

Do you happen to know what I am doing wrong , 
Thank you very much

**Comment by lalalagartija** [comment, marked as answer]
Hi ! 
I solved my issue. There was womething wrong with my bed file then I regenerated it and I added rgb() to the color column. 
The command line I did was : 
jbrowse add-track blast_clusters_simple.bb -a Virus1 --load copy --out /var/www/html/viruses/ --force --config '{"displays": [{"displayId": "basicdisplay","type":"LinearBasicDisplay", "renderer": {"color1": "jexl:get(feature, @thecolor@)", "type": "SvgFeatureRenderer"}}]}' 

then 
sed -i "s/@thecolor@/\'thecolor\'/g" /var/www/html/viruses/config.json
